src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java

```diff
@@ -1725,7 +1746,22 @@ public void updateTriggerConfigURL() {
                         + "a specific configured server");
             } else {
                 List<GerritProject> fetchedProjects = GerritDynamicUrlProcessor.fetch(triggerConfigURL, serverName);
-                dynamicGerritProjects = fetchedProjects;
+                // Since other threads may be accessing the dynamic project list, we need to make sure
+                // that we update its contents in a thread-safe way.
```

Comment:
Sorry, can't get why it was not thread-safe before?

Reply:
You're right; I had originally tried to lock on `dynamicGerritProjects`, but since it could be null, that was a bad idea.  Since the address of `dyanmicGerritProjects` isn't used for any reason (for example, locking), we can probably go back to the switcheroo here.

Reply:
This has been reverted.